common/ratelimit/limiter.go

```diff
@@ -109,7 +125,44 @@ func (d *rateLimiter) checkAllowed(ctx context.Context, params common.RequestPar
 		bucketParams.BucketLevels[i] = getBucketLevel(bucketParams.BucketLevels[i], size, interval, deduction)
 		allowed = allowed && bucketParams.BucketLevels[i] > 0
 
-		d.logger.Debug("Bucket level", "key", params.RequesterID, "prevLevel", prevLevel, "level", bucketParams.BucketLevels[i], "size", size, "interval", interval, "deduction", deduction, "allowed", allowed)
+		if strings.HasPrefix(params.RequesterID, "system:") {
```

Comment:
This logic seems pretty highly coupled with the format of the keys that are passed into the library methods, which someone could change. I think a simpler interface--even if it results in less intricate metrics--is probably preferable.

A suggestion would be to replace `params.IsAuthenticated` with a string parameter `params.MetricsKey`. If the string is blank, no metrics are recorded. If not, the metrics are recorded with this key. What do you think?

Reply:
So are you thinking we would just add a single label called "metrics_key" and pass in the requesterId?

Reply:
I could see having two levels. Maybe something like
- `MetricsPrimaryKey`
- `MetricsSecondaryKey`

or 
- `MetricsKey`
- `MetricsTags`

Then the user of the library could populate these as desired.